clash-prelude/src/Clash/Sized/Vector.hs

```diff
@@ -1111,6 +1112,11 @@ scanl f z xs = ws
     ws = z `Cons` zipWith (flip f) xs (init ws)
 {-# INLINE scanl #-}
 
+-- | 'scanl' with no seed value
+scanl1 :: KnownNat n => (a -> a -> a) -> Vec (n+1) a -> Vec (n+1) a
```

Comment:
I think if we add `scanl1` to `clash-prelude` we should also add `scanr1` for completeness

Reply:
Yes, fair enough.